### PR TITLE
BUILD-3399 Update Maven plugins

### PR DESCRIPTION
**Pull request by henryju**
Big batch of updates, I will test that on a few projects.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -130,32 +129,13 @@
   </properties>
 
   <build>
-    <extensions>
-      <!-- See SONARPLUGINS-839 and SONARPLUGINS-840 -->
-      <extension>
-        <groupId>org.apache.maven.wagon</groupId>
-        <artifactId>wagon-webdav</artifactId>
```
Comment (henryju) [inline pom.xml:137]: Deprecated, will be removed in Maven 4

File: pom.xml [hunk 2]
```diff
@@ -235,11 +215,6 @@
             </licenseMerges>
           </configuration>
         </plugin>
-        <plugin>
-          <groupId>org.sonatype.plugins</groupId>
-          <artifactId>jarjar-maven-plugin</artifactId>
```
Comment (henryju) [inline pom.xml:240]: This plugin is discontinued, I don't think the parent pom should encourage its use. If some projects are using it, they should probably move to the Maven Shade plugin.

Comment (julien-carsique-sonarsource) [reply pom.xml:240]: This would break only sonar-pli if also applied to parent

File: pom.xml [hunk 3]
```diff
@@ -87,37 +87,36 @@
...
     <version.sonar-packaging.plugin>1.21.0.505</version.sonar-packaging.plugin>
     <version.sonar-dev.plugin>1.8</version.sonar-dev.plugin>
 
-    <version.codehaus.license.plugin>2.0.0</version.codehaus.license.plugin>
-    <version.mycila.license.plugin>3.0</version.mycila.license.plugin>
-    <version.gpg.plugin>0.3.1</version.gpg.plugin>
-    <version.cyclonedx.plugin>2.6.0</version.cyclonedx.plugin>
+    <version.codehaus.license.plugin>2.2.0</version.codehaus.license.plugin>
+    <version.mycila.license.plugin>4.2</version.mycila.license.plugin>
+    <version.gpg.plugin>3.1.0</version.gpg.plugin>
```
Comment (julien-carsique-sonarsource) [inline pom.xml:118]:
```suggestion
    <version.gpg.plugin>1.0.1</version.gpg.plugin>
```

File: pom.xml [hunk 4]
```diff
@@ -87,37 +87,36 @@
...
+    <version.jar.plugin>3.3.0</version.jar.plugin>
+    <version.javadoc.plugin>3.5.0</version.javadoc.plugin>
+    <version.plugin.plugin>3.9.0</version.plugin.plugin>
+    <version.resources.plugin>3.3.1</version.resources.plugin>
+    <version.shade.plugin>3.5.0</version.shade.plugin>
+    <version.source.plugin>3.3.0</version.source.plugin>
     <version.site.plugin>3.7.1</version.site.plugin>
-    <version.artifactory.plugin>3.4.0</version.artifactory.plugin>
-    <version.jacoco.plugin>0.8.8</version.jacoco.plugin>
+    <version.artifactory.plugin>3.6.1</version.artifactory.plugin>
```
Comment (julien-carsique-sonarsource) [inline pom.xml]: This is a breaking change

Comment (henryju) [reply pom.xml]: Should a separate ticket be created to investigate what should be fixed? Not being able to update a plugin is dangerous in the long run.

Comment (julien-carsique-sonarsource) [reply pom.xml]: For artifactory, yes this needs dedicated work as it's widely used and they apparently changed some behaviors with the environment variables.

File: pom.xml [hunk 5]
```diff
@@ -87,37 +87,36 @@
...
+    <version.shade.plugin>3.5.0</version.shade.plugin>
+    <version.source.plugin>3.3.0</version.source.plugin>
     <version.site.plugin>3.7.1</version.site.plugin>
     <version.artifactory.plugin>3.4.0</version.artifactory.plugin>
-    <version.jacoco.plugin>0.8.8</version.jacoco.plugin>
+    <version.jacoco.plugin>0.8.10</version.jacoco.plugin>
 
-    <version.buildnumber.plugin>1.4</version.buildnumber.plugin>
-    <version.native2ascii.plugin>1.0-beta-1</version.native2ascii.plugin>
+    <version.buildnumber.plugin>3.1.0</version.buildnumber.plugin>
```
Comment (julien-carsique-sonarsource) [inline pom.xml:110]: there's a 3.2.0 available

**Review by julien-carsique-sonarsource — APPROVED** [review]

File: pom.xml (hunk 5 above)
Comment (henryju) [reply pom.xml:110]: The [website](https://www.mojohaus.org/buildnumber-maven-plugin/) is not updated, and the [changes](https://github.com/mojohaus/buildnumber-maven-plugin/releases/tag/buildnumber-maven-plugin-3.2.0) are pretty minor, so I think we can wait for the next time.

**Pull request merged**